src/sys/time.rs

```diff
@@ -5,6 +5,130 @@ use libc::{timespec, timeval};
 #[cfg_attr(target_env = "musl", allow(deprecated))] // https://github.com/rust-lang/libc/issues/1848
 pub use libc::{time_t, suseconds_t};
 
+#[cfg(any(
+    all(feature = "time", any(target_os = "android", target_os = "linux")),
+    all(
+        any(
+            target_os = "android",
```

Comment:
Should Android really be here, even though it's also on line 9?

Reply:
I think there's probably a handful of different ways to write this logic. Is there an alternate form you would prefer to see?

Reply:
Why is it that the signal feature is only required on BSD and illumos, anyway?

Reply:
Right, yeah. So the `timer` submodule here is required by both the `timer` and `timerfd` mechanism. The `timerfd` bit of the cfg here is what you see on line 9, the `timer` bit starts on 10. Android could actually be removed from the part that starts on 10. The need for signal is a consequence of something hack turned up. If we flag this whole common module on by OS if signal is flagged off then `TimerSpec::none` is unused on the BSDs.

Reply:
Ok.  Weird, but I get it.